src/metabase/permissions/models/data_permissions.clj

```diff
@@ -139,6 +139,22 @@
   When checking permissions, if a DB has not been fetched, it will be added to the cache before the check returns."
   (atom {:db-ids #{} :perms {}}))
 
+(defn prime-db-cache
```

Comment:
Maybe worth having a unit test for this function? or maybe just a test asserting on the number of DB calls that are made when calling `/api/database` to ensure that's priming the cache correctly? there are a couple examples in `metabase.models.data-permissions-test` already that test perms caching logic using `t2/with-call-count`

Reply:
I had seen with-call-count and had tried adding that to the `api.database-test` but it ends up being a "assert the query count is 8" which seemed sort of a random number which can fail for other changes that may happen over time in the API and not well capturing exactly what's happening. 

But, it does make sure it doesn't get excessively large. So I can add it back if you think it's best